gamersnet_backend/routes/posts/getPosts.js

```diff
@@ -1,10 +1,44 @@
 'use strict';
 
 
-// include our function from the database to add post
-let {getAllPosts, getValidPosts} = require('../../persistence/posts');
+// include our function from the database to fetch posts
+let {getUserPosts, getAllPosts, getValidPosts} = require('../../persistence/posts');
 
-// this function handles the /post/getAllPosts/ endpoint
+let {verifyUserLoggedIn} = require('../utilities/tokenUtility')
+
+let {getUserIDFromToken} = require('../../persistence/tokens.js')
+
+// this function handles the /posts/listUserPosts/ endpoint
+async function listAllUserPosts(request, response) {
+    let cookie = request.headers.cookie;
+
+    let loggedIn = false;
+
+    if (cookie) {
+        cookie = cookie.split('=')[1];
+        loggedIn = await verifyUserLoggedIn(cookie);
+    }
+
+    if (loggedIn){
+        let tokenDocument = await getUserIDFromToken(cookie);
+
+        //input type are verified here
+        let userID = tokenDocument.userID;
```

Comment:
I think you need to get the used id from the request body instead of the token since you will be passing the userid from the frontend and it may not be same as the user id in the token. you just need the token to verify login

Reply:
This is just to display posts for the user that's logged in and that's why it's taking it from the token and not passing in from the frontend

Reply:
Oh sorry i didn't notice the description. In that case name the endpoint more appropriately like listLoggedUserPosts or something like that(If you don't plan to reuse it for userids in general).. so that it's not misleading.

Reply:
Give a quick look; passed userID in the body and empty string denotes the logged in user has requested list of Posts.